### PR TITLE
Add java tools sources archive and update the deploy jars.

### PR DESCRIPTION
**Pull request by iirina**
* Updated `third_party/java/java_tools/update_java_tools.sh` to create a zip of the java tools source code.
* Run `third_party/java/java_tools/update_java_tools.sh` to update all the tools and create the initial source zip.
* Added a target for creating the Java tools archive that will be uploaded to `mirror.bazel.build`.

### CODE REVIEW

File: third_party/java/java_tools/update_java_tools.sh [hunk 1]
```diff
@@ -24,6 +24,11 @@ usage="This script updates the checked-in jars corresponding to the tools "\
 To update all the tools simultaneously run from your bazel workspace root:
 ~/third_party/java/java_tools/update_java_tools.sh
```
Comment (meisterT) [inline third_party/java/java_tools/update_java_tools.sh]: Nit: add "WARNING:"

File: third_party/java/java_tools/update_java_tools.sh [hunk 2]
```diff
@@ -43,18 +48,30 @@ fi
...
 if [[ ! -z "$@" ]]
 then
    # Update only the tools specified on the command line.
    tools_to_update=("$@")
+   echo "The sources archive was NOT created. Please create it manually!"
 else
   # If no tools were specified update all of them.
   tools_to_update=(${!tool_name_to_target[*]})
+  # Create the sources archive only when all the tools were updated.
+  zip -r third_party/java/java_tools/java_tools-srcs.zip src/java_tools/buildjar/* \
```
Comment (meisterT) [inline third_party/java/java_tools/update_java_tools.sh]: This is non-deterministic, can you please try to remove the timestamps, so it does give the same md5 sum every time?

File: third_party/java/java_tools/update_java_tools.sh (hunk 1 above)
Comment (iirina) [reply third_party/java/java_tools/update_java_tools.sh]: Done.

File: third_party/java/java_tools/update_java_tools.sh (hunk 2 above)
Comment (iirina) [reply third_party/java/java_tools/update_java_tools.sh]: Good point. I added `-X` option which doesn't include file times

File: third_party/java/java_tools/update_java_tools.sh [hunk 3]
```diff
@@ -62,7 +62,8 @@ else
   # If no tools were specified update all of them.
   tools_to_update=(${!tool_name_to_target[*]})
   # Create the sources archive only when all the tools were updated.
-  zip -r third_party/java/java_tools/java_tools-srcs.zip src/java_tools/buildjar/* \
+  # -X doesn't save extra file attributes (e.g. file times)
+  zip -Xr third_party/java/java_tools/java_tools-srcs.zip src/java_tools/buildjar/* \
```
Comment (meisterT) [inline third_party/java/java_tools/update_java_tools.sh]: As discussed offline, I don't think this is enough to remove all the times. But feel free to move and add a TODO here instead.

Comment (lberki) [reply third_party/java/java_tools/update_java_tools.sh]: Yeah. If you want to emulate what we are doing with the actual Bazel zip, you'll end up writing a Python script:

https://github.com/bazelbuild/bazel/blob/master/src/create_embedded_tools.py#L133

That would also be consistent rule of thumb of "if your shell script is more than three lines long, write it in Python" :)

I don't know by heart what kind of zip handling tools we have in this repository, but it's conceivable that something can also be jury-rigged from the code we use for ijar/singlejar.

I also agree with @meisterT that this is fine for the time being. Having a zip with time stamps beats not having a zip :)

**Comment by iirina** [comment]
Merged in 73d5381dc2c5b4df92ddcf9b03a486752cb43caf.

**Pull request closed without merging**